smithy-typescript-codegen/src/main/java/software/amazon/smithy/typescript/codegen/ServiceGenerator.java

```diff
@@ -313,7 +314,9 @@ private void generateConstructor() {
             for (RuntimeClientPlugin plugin : runtimePlugins) {
                 if (plugin.getResolveFunction().isPresent()) {
                     configVariable++;
-                    List<String> additionalParameters = plugin.getAdditionalResolveFunctionParameters();
+                    Map<String, Object> paramsMap = plugin.getResolveFunctionParameters(model, service, null);
+                    List<String> additionalParameters = CodegenUtils.getFunctionParametersList(paramsMap);
```

Comment:
Why can't `getResolveFunctionParameters()` returns `additionalParameters` directly? `CodegenUtils.getFunctionParametersList` is only used here and doesn't require more parameters. I feel like it's not necessary.